realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1302,4 +1411,112 @@ public E findFirst() {
             return null;
         }
     }
+
+    /**
+     * Represents a pending asynchronous Realm query.
+     *
+     * Users are responsible of maintaining a reference to {@code AsyncRequest} in order
+     * to call #cancel in case of a configuration change for example (to avoid memory leak, as the
+     * query will post the result to the caller's thread callback)
+     */
+    public class AsyncRequest {
```

Comment:
Should we just call this Request instead? You would normally use it like new RealmQuery.Request(callback), which would imply async behaviour anyway?